pulser-core/pulser/noise_model.py

```diff
@@ -408,16 +410,16 @@ def _validate_parameters(param_vals: dict[str, Any]) -> None:
                 )
 
     def _to_abstract_repr(self) -> dict[str, Any]:
-        all_fields = asdict(self)
+        all_fields = {}
+        for f in fields(self):
+            value = getattr(self, f.name)
+            if f.name in OPTIONAL_IN_ABSTR_REPR and f.default == value:
```

Comment:
Not relevant for now but this will fail if the field uses a `default_factory` instead of a `default`. You can use `pulser.json.utils.get_dataclass_defaults()` instead